### PR TITLE
location of user implemented

### PR DESCRIPTION
**Pull request by batchatT**

### CODE REVIEW

File: app/src/main/java/com/example/trempelapp/presentation_layer/main_screen/main_fragments/profile/FusedLocationProviderManager.kt [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        locationRequest = LocationRequest.create()
+        locationRequest?.interval = 5000
+        locationRequest?.fastestInterval = 5000
+        locationRequest?.priority = LocationRequest.PRIORITY_HIGH_ACCURACY
+
+        locationCallback = object : LocationCallback() {
+            override fun onLocationResult(locationResult: LocationResult) {
+                super.onLocationResult(locationResult)
+                val location = locationResult.locations.last()
+                Log.d(
```
Comment (denisque75) [inline app/src/main/java/com/example/trempelapp/presentation_layer/main_screen/main_fragments/profile/FusedLocationProviderManager.kt]: logs are redundant

File: app/src/main/java/com/example/trempelapp/presentation_layer/main_screen/main_fragments/profile/FusedLocationProviderManager.kt [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+    @SuppressLint("MissingPermission")
+    fun getLocation(profileState: SingleLiveEvent<ProfileState>) {
+        fusedLocationProviderClient?.lastLocation?.addOnSuccessListener() {
+            it?.let {
+                val currentLocation = it
+                profileState.value = profileState.value?.copy(
+                    location = UserLocation(currentLocation.latitude, currentLocation.longitude)
+                )
+                Log.d(
```
Comment (denisque75) [inline app/src/main/java/com/example/trempelapp/presentation_layer/main_screen/main_fragments/profile/FusedLocationProviderManager.kt]: logs are redundant

File: app/src/main/java/com/example/trempelapp/presentation_layer/main_screen/main_fragments/profile/FusedLocationProviderManager.kt [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+                    "locationtag",
+                    "${currentLocation.latitude} + \",\" +\n${currentLocation.longitude}"
+                )
+            } ?: fusedLocationProviderClient?.requestLocationUpdates(
+                locationRequest!!,
```
Comment (denisque75) [inline app/src/main/java/com/example/trempelapp/presentation_layer/main_screen/main_fragments/profile/FusedLocationProviderManager.kt]: avoid non null assertion

File: app/src/main/java/com/example/trempelapp/presentation_layer/main_screen/main_fragments/profile/FusedLocationProviderManager.kt [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+                locationCallback!!,
```
Comment (denisque75) [inline app/src/main/java/com/example/trempelapp/presentation_layer/main_screen/main_fragments/profile/FusedLocationProviderManager.kt]: avoid non null assertion

File: app/src/main/java/com/example/trempelapp/presentation_layer/main_screen/main_fragments/profile/ProfilePageFragment.kt [hunk 5]
```diff
@@ -120,34 +169,135 @@ class ProfilePageFragment : BaseFragment() {
...
+    @Composable
+    private fun CheckPermissionStatus(
+        perLauncher: ActivityResultLauncher<Array<String>>,
+        latitude: Double?,
+        longitude: Double?
+    ) {
+        val finePermissionStatus =
+            ContextCompat.checkSelfPermission(requireContext(), fineLocationPermission)
+        if (viewModel.gpsStatusLiveData.value == true) {
+            if (finePermissionStatus == -1) {
```
Comment (denisque75) [inline app/src/main/java/com/example/trempelapp/presentation_layer/main_screen/main_fragments/profile/ProfilePageFragment.kt]: use constant, to add readability

File: app/src/main/java/com/example/trempelapp/presentation_layer/main_screen/main_fragments/profile/ProfilePageFragment.kt [hunk 6]
```diff
@@ -120,34 +169,135 @@ class ProfilePageFragment : BaseFragment() {
...
+
+    @Composable
+    private fun SetMapWithoutPermission(error: String) {
+        Box(
+            modifier = Modifier
+                .padding(start = 40.dp, end = 40.dp)
+                .clip(RoundedCornerShape(10)),
+        ) {
+
+            val singapore = LatLng(1.35, 103.87)
```
Comment (denisque75) [inline app/src/main/java/com/example/trempelapp/presentation_layer/main_screen/main_fragments/profile/ProfilePageFragment.kt]: make it as constant

File: app/src/main/java/com/example/trempelapp/presentation_layer/main_screen/main_fragments/profile/ProfilePageFragment.kt [hunk 7]
```diff
@@ -120,34 +169,135 @@ class ProfilePageFragment : BaseFragment() {
...
+        ) {
+
+            val singapore = remember {
+                mutableStateOf<LatLng?>(null)
+            }
+
+            singapore.value = if (latitude != null && longitude != null) {
+                LatLng(latitude, longitude)
+            } else {
+                LatLng(1.35, 103.87)
```
Comment (denisque75) [inline app/src/main/java/com/example/trempelapp/presentation_layer/main_screen/main_fragments/profile/ProfilePageFragment.kt]: use constant

File: app/src/main/java/com/example/trempelapp/presentation_layer/main_screen/main_fragments/profile/FusedLocationProviderManager.kt (hunk 2 above)
Comment (batchatT) [reply app/src/main/java/com/example/trempelapp/presentation_layer/main_screen/main_fragments/profile/FusedLocationProviderManager.kt]: done

File: app/src/main/java/com/example/trempelapp/presentation_layer/main_screen/main_fragments/profile/FusedLocationProviderManager.kt (hunk 3 above)
Comment (batchatT) [reply app/src/main/java/com/example/trempelapp/presentation_layer/main_screen/main_fragments/profile/FusedLocationProviderManager.kt]: done

File: app/src/main/java/com/example/trempelapp/presentation_layer/main_screen/main_fragments/profile/FusedLocationProviderManager.kt (hunk 4 above)
Comment (batchatT) [reply app/src/main/java/com/example/trempelapp/presentation_layer/main_screen/main_fragments/profile/FusedLocationProviderManager.kt]: done

File: app/src/main/java/com/example/trempelapp/presentation_layer/main_screen/main_fragments/profile/ProfilePageFragment.kt (hunk 5 above)
Comment (batchatT) [reply app/src/main/java/com/example/trempelapp/presentation_layer/main_screen/main_fragments/profile/ProfilePageFragment.kt]: done

File: app/src/main/java/com/example/trempelapp/presentation_layer/main_screen/main_fragments/profile/ProfilePageFragment.kt (hunk 6 above)
Comment (batchatT) [reply app/src/main/java/com/example/trempelapp/presentation_layer/main_screen/main_fragments/profile/ProfilePageFragment.kt]: done

File: app/src/main/java/com/example/trempelapp/presentation_layer/main_screen/main_fragments/profile/ProfilePageFragment.kt (hunk 7 above)
Comment (batchatT) [reply app/src/main/java/com/example/trempelapp/presentation_layer/main_screen/main_fragments/profile/ProfilePageFragment.kt]: done

**Pull request merged**